src/components/view/transaction/transaction-contract-details/TransactionContractDetails.tsx

```diff
@@ -59,7 +59,14 @@ export const TransactionContractDetails: React.FC<{
       case "/vm.m_addpkg":
         return "AddPackage";
       case "/vm.m_call":
-        return message["func"] || message["@type"];
+        const arg0 = message["args"] && message["args"][0];
+        if (typeof arg0 === "string" && arg0.trim() !== "") {
+          return arg0;
+        }
+        if (typeof message["func"] === "string" && message["func"].trim() !== "") {
+          return message["func"];
+        }
+        return message["@type"];
```

Comment:
It would be safer to first validate arrays with `Array.isArray` and then check the `func`.